docker/dotnet/Dockerfile

```diff
@@ -3,12 +3,18 @@
 #   if unset then it uses the build generated above. (TODO: to be done)
 # DOTNET_AGENT_REPO and DOTNET_AGENT_BRANCH parameterise the DOTNET agent
 # repo and branch (or commit) to use.
-FROM mcr.microsoft.com/dotnet/core/sdk:3.1.100 AS build
+FROM mcr.microsoft.com/dotnet/sdk:5.0 AS build
+ENV DOTNET_ROOT=/usr/share/dotnet
 ARG DOTNET_AGENT_REPO=elastic/apm-agent-dotnet
 ARG DOTNET_AGENT_BRANCH=master
 ARG DOTNET_AGENT_VERSION=
+# Workaround for https://github.com/dotnet/sdk/issues/14497
+ARG DOTNET_HOST_PATH=/usr/share/dotnet/dotnet
 WORKDIR /src
 COPY . /src
+# install SDK version in global.json of elastic/apm-agent-dotnet
+# TODO: remove once global.json is on 5.0 *and* PRs are updated to 5.0 in global.json
```

Comment:
I guess for backward compatibility reasons we might need to keep it

```suggestion
# Backward compatibility when global.json is *not* on 5.0
```

Reply:
This is only needed whilst there are branches that build on CI that still specify `3.1.100` as the SDK to use in global.json. At some point in the near future after this PR is merged and global.json is updated to `5.0.100` SDK in https://github.com/elastic/apm-agent-dotnet/pull/1040, existing open PRs specifying `3.1.100` can be rebased to bring in the change to `5.0.100`, and new PRs will specify `5.0.100` as the SDK to use. From then, I don't think we need to install `3.1.100` SDK.

Reply:
I've updated the comments